Sources/ResilientDecoding/ErrorReporting.swift

```diff
@@ -132,6 +141,30 @@ public struct ErrorDigest {
 
 }
 
+// MARK: - Reporting Errors
+
+extension Decoder {
+
+  /**
+   This method should be called whenever an error is handled by the `Resilient` infrastructure.
+   Care should be taken that this is called on the most relevant `Decoder` object, since this method uses the `Decoder`'s `codingPath` to place the error in the correct location in the tree.
+   */
+  func resilientDecodingHandled(_ error: Swift.Error) {
+    guard let errorReporterAny = userInfo[.resilientDecodingErrorReporter] else {
+      return
+    }
+    /**
+     Check that we haven't hit the very unlikely case where someone has overriden our user info key with something we do not expect.
+     */
+    guard let errorReporter = errorReporterAny as? ResilientDecodingErrorReporter else {
+      assertionFailure()
```

Comment:
maybe include a message with the failure?

Reply:
This is the pattern we've been following... since this project is meant to always be included as source code I think its safe to just rely on the comments around the assertion.